src/WrappedMToken.sol

```diff
@@ -81,10 +81,12 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
     /// @inheritdoc IWrappedMToken
     function wrap(address recipient_, uint256 amount_) external {
+        uint256 startingBalance_ = IMTokenLike(mToken).balanceOf(address(this));
+
         // NOTE: The behavior of `IMTokenLike.transferFrom` is known, so its return can be ignored.
         IMTokenLike(mToken).transferFrom(msg.sender, address(this), amount_);
 
-        _mint(recipient_, UIntMath.safe240(amount_));
+        _mint(recipient_, UIntMath.safe240(IMTokenLike(mToken).balanceOf(address(this)) - startingBalance_));
```

Comment:
todo add a note about why diff is used with a situation of transfer between non-earner to earner (this wrapper)